lib/src/core/routes/route_transitions.dart

```diff
@@ -135,9 +126,13 @@ class RouteTransitionBuilder<T> extends RouteTransition<T> {
 
 /// Represents a route transition with various [transitionSettings] to its parameters.
 abstract class RouteTransition<T> extends PageRoute<T> {
-  RouteTransition({RouteSettings? settings, RouteTransitionSettings? transitionSettings})
-    : transitionSettings = transitionSettings ?? RouteTransitionSettings(),
-      super(settings: settings);
+  RouteTransition({
+    super.settings,
+    RouteTransitionSettings? transitionSettings,
+    super.barrierDismissible,
+    this.barrierColor,
+    this.barrierLabel,
```

Comment:
Overwriting the final attributes in subclasses generated a note, so I've allowed them to be set via the constructor.
This is not a breaking change, since both of them are optional.